coding/python_testing.rst

```diff
@@ -4,6 +4,204 @@ Python Unit Testing
+   collected 1 items
+
+   coding/unit_test_snippets/unittest_baseclass.py::ThisIsTest1::testParam PASSED
+
+   ===================================== 1 passed in 0.02 seconds =====================================
+
+
+LSST Utility Test Support Classes
+=================================
+
+:lmod:`lsst.utils.tests` provides several utilities functions and classes for writing Python tests that developers should make use of.
+
+.. _py-utils-tests:
+
+Special Asserts
+---------------
+
+Inheriting from :lclass:`lsst.utils.tests.TestCase` rather than :class:`unittest.TestCase` enables two new asserts that are useful for doing element-wise comparison of two :mod:`numpy` arrays.
+
+:lmeth:`~lsst.utils.tests.TestCase.assertClose`
+   Asserts that the corresponding elements in two arrays are within the specified tolerance.
+:lmeth:`~lsst.utils.tests.TestCase.assertNotClose`
+   Asserts that the two arrays are different.
+
+Additionally, :lmod:`~lsst.afw` provides additional asserts that get loaded into :lclass:`lsst.utils.tests.TestCase` when the associated module is loaded.
+
+:lmeth:`~lsst.afw.coord.utils.assertCoordsNearlyEqual`
+   Assert that two coords represent nearly the same point on the sky (provided by :lmod:`lsst.afw.coord.utils`).
+:lmeth:`~lsst.afw.geom.utils.assertAnglesNearlyEqual`
+   Assert that two angles are nearly equal, ignoring wrap differences by default (provided by :lmod:`lsst.afw.geom.utils`).
+:lmeth:`~lsst.afw.geom.utils.assertPairsNearlyEqual`
+   Assert that two planar pairs (e.g. :lclass:`~lsst.afw.geom.Point2D` or :lclass:`~lsst.afw.geom.Extent2D`) are nearly equal (provided by :lmod:`lsst.afw.geom.utils`).
+:lmeth:`~lsst.afw.geom.utils.assertBoxesNearlyEqual`
+   Assert that two boxes (:lclass:`~lsst.afw.geom.Box2D` or :lclass:`~lsst.afw.geom.Box2I`) are nearly equal (provided by :lmod:`lsst.afw.geom.utils`).
+:lmeth:`~lsst.afw.image.basicUtils.assertWcsNearlyEqualOverBBox`
+   Compare :lmeth:`~lsst.afw.image.imageLib.Wcs.pixelToSky` and :lmeth:`~lsst.afw.image.imageLib.Wcs.skyToPixel` for two WCS over a rectangular grid of pixel positions (provided by :lmod:`lsst.afw.image.basicUtils`).
+:lmeth:`~lsst.afw.image.testUtils.assertImagesNearlyEqual`
+   Assert that two images are nearly equal, including non-finite values (provided by :lmod:`lsst.afw.image.testUtils`).
+:lmeth:`~lsst.afw.image.testUtils.assertMasksEqual`
+   Assert that two masks are equal (provided by :lmod:`lsst.afw.image.testUtils`).
+:lmeth:`~lsst.afw.image.testUtils.assertMaskedImagesNearlyEqual`
+   Assert that two masked images are nearly equal, including non-finite values (provided by :lmod:`lsst.afw.image.testUtils`).
+
+Testing Executables
+-------------------
+
+In some cases the test to be executed is a shell script or a compiled binary executable.
+In order for the test running environment to be aware of these tests a Python test file must be present that can be run by `pytest`_ to executes these tests in the :mod:`unittest` environment.
```

Comment:
Semi-optional comma between "tests" and "a" to break up the long sentence.  "executes" -> "execute".  "in the unittest environment" is somewhat redundant with "In order for the test running environment to be aware".